### PR TITLE
bpo-34266: [pdb] handle ValueError from shlex.split()

### PR DESCRIPTION
**Pull request by iritkatriel**
https://bugs.python.org/issue34266

### CODE REVIEW

**Comment by iritkatriel** [comment]
@blueyed Are you able to review?

File: Misc/NEWS.d/next/Library/2021-06-10-21-53-46.bpo-34266.k3fxnm.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Handle exceptions from parsing the arg of mod:`pdb`'s run/restart command.
```
Comment (jacobtylerwalls) [inline Misc/NEWS.d/next/Library/2021-06-10-21-53-46.bpo-34266.k3fxnm.rst]:
```suggestion
Handle exceptions from parsing the arg of :mod:`pdb`'s run/restart command.
```

File: Lib/pdb.py [hunk 2]
```diff
@@ -1026,7 +1026,11 @@ def do_run(self, arg):
         if arg:
             import shlex
             argv0 = sys.argv[0:1]
-            sys.argv = shlex.split(arg)
+            try:
+                sys.argv = shlex.split(arg)
+            except ValueError:
+                self.error('Cannot run: %s' % arg)
```
Comment (serhiy-storchaka) [inline Lib/pdb.py]: Would not be helpful to show also the shlex error message?

File: Lib/test/test_pdb.py [hunk 3]
```diff
@@ -1771,6 +1771,18 @@ def test_errors_in_command(self):
             '(Pdb) ',
         ])
 
+    def test_issue34266(self):
+        '''do_run handles exceptions from parsing its arg'''
+        commands = "\n".join([
+            'run \\',
```
Comment (serhiy-storchaka) [inline Lib/test/test_pdb.py]: It is worth to add also a test for `run "`. There are only to causes of ValueError in `shlex.split()`. Interesting, that gdb handles `r \` and `r "` in different ways, so it is possible that we will differentiate them too in future.

File: Lib/test/test_pdb.py [hunk 4]
```diff
@@ -1773,16 +1773,19 @@ def test_errors_in_command(self):
...
+                'q',
+            ])
+            stdout, _ = self.run_pdb_script('pass', commands + '\n')
+            self.assertEqual(stdout.splitlines()[1:], [
+                '-> pass',
+                f'(Pdb) *** Cannot run {bad_arg}: {msg}',
+                '(Pdb) ',
+            ])
+        check('\\', 'No escaped character')
+        check('\"', 'No closing quotation')
```
Comment (serhiy-storchaka) [inline Lib/test/test_pdb.py]: `'\"'` is the same as `'"'`.

File: Lib/test/test_pdb.py [hunk 5]
```diff
@@ -1771,6 +1771,21 @@ def test_errors_in_command(self):
...
```
Comment (iritkatriel) [inline Lib/test/test_pdb.py]:
```suggestion
        check('"', 'No closing quotation')
```

**Review by serhiy-storchaka — APPROVED** [review]
LGTM.

**Pull request merged**